src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const finalStates = [];
+  const stateClone = {};
+
+  Object.assign(stateClone, state);
+
+  for (const step in actions) {
+    switch (actions[step].type) {
+      case 'addProperties':
+        Object.assign(stateClone, actions[step].extraData);
+        break;
+      case 'removeProperties':
+        for (const key in actions[step].keysToRemove) {
+          const removeKey = actions[step].keysToRemove[key];
+
+          delete stateClone[removeKey];
+        }
+        break;
+
+      default:
+        for (const key in stateClone) {
+          delete stateClone[key];
+        }
```

Comment:
The default case currently deletes all properties from `stateClone`. Ensure this behavior is intended, as it will result in an empty state object for any unrecognized action type.